lib/srv/db/snowflake/cache.go

```diff
@@ -0,0 +1,41 @@
+/*
+
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+
+*/
+
+package snowflake
+
+// tokenCache is a simple in memory map between Teleport WebSession ID and Snowflake tokens.
+type tokenCache struct {
+	tokens map[string]string
+}
+
+func newTokenCache() tokenCache {
+	return tokenCache{tokens: make(map[string]string)}
+}
+
+func (t *tokenCache) getToken(sessionID string) string {
+	return t.tokens[sessionID]
```

Comment:
Potential race condition ?

Reply:
I don't think so. Cache is created by connection and it's not shared with anyone else. Each connection run only in one goroutine, so this call is safe.